src/pages/stacks-blockchain/integrate-stacking.md

```diff
@@ -0,0 +1,366 @@
+
+console.log({
+  stackingExecution,
+  cycleDuration,
+  nextCycleStartingAt,
+  // mimnimum micro-Stacks required to participate
+  minimumUSTX: poxInfo.min_amount_ustx,
+});
+```
+
+Users need to have sufficient Stacks (STX) tokens to participate in Stacking. With the Stacking info, this can be verified easily:
+
+```js
+const accounts = new AccountsApi(apiConfig);
+
+const accountBalance = await accounts.getAccountBalance({
+  principal: stxAddress,
+});
+
+const accountSTXBalance = accountBalance.stx.balance;
+
+// enough balance for participation?
+const canParticipate = accountSTXBalance >= poxInfo.min_amount_ustx;
+
+console.log({
+  stxAddress,
+  accountSTXBalance,
+  canParticipate,
+});
+```
+
+For testing purposes, you can use a faucet to obtain STX tokens:
+
+```shell
+curl -XPOST "https://stacks-node-api.blockstack.org/extended/v1/faucets/stx?address=<stxAddress>&stacking=true"
+```
+
+You will have to wait a few minutes for the transaction to complete.
+
+Users can select how many cycles they would like to participate in. To help with that decision, the unlocking time can be estimated:
+
+```js
+// this would be provided by the user
+let numberOfCycles = 3;
+
+// the projected datetime for the unlocking of tokens
+const unlockingAt = new Date(nextCycleStartingAt);
+unlockingAt.setSeconds(
+  unlockingAt.getSeconds() +
+    poxInfo.reward_cycle_length * numberOfCycles * blocktimeInfo.testnet.target_block_time
+);
+```
+
+## Step 4: Verify stacking eligibility
+
+At this point, your app shows Stacking details. If Stacking is executed and the user has enough funds, the user should be asked to provide input for the amount of micro-STX to lockup and a bitcoin address to be used to pay out rewards.
+
+-> The sample code used assumes usage of the bitcoin address associated with the Stacks account. You can replace this with an address provided by the users or read from your database
+
+With this input, and the data from previous steps, we can determine the eligibility for the next reward cycle:
+
+```js
+// micro-STX tokens to lockup, must be >= poxInfo.min_amount_ustx and <=accountSTXBalance
+let microSTXoLockup = poxInfo.min_amount_ustx;
+
+// derive bitcoin address from Stacks account and convert into required format
+const hashbytes = bufferCV(Buffer.from(c32.c32addressDecode(stxAddress)[1], 'hex'));
+const version = bufferCV(Buffer.from('01', 'hex'));
+
+const smartContracts = new SmartContractsApi(apiConfig);
+
+// read-only contract call
+const isEligible = await smartContracts.callReadOnlyFunction({
+  contractAddress: poxInfo.contract_id.split('.')[0],
+  contractName: poxInfo.contract_id.split('.')[1],
+  functionName: 'can-stack-stx',
+  readOnlyFunctionArgs: {
+    sender: stxAddress,
+    arguments: [
+      `0x${serializeCV(
+        tupleCV({
+          hashbytes,
+          version,
+        })
+      ).toString('hex')}`,
+      `0x${serializeCV(uintCV(microSTXoLockup)).toString('hex')}`,
```

Comment:
We should definitely add some helper methods around this.

Reply:
in this sample or in our tooling?

Reply:
in the tooling

Reply:
Yes, agreed! It's in my list of suggestions!